File: OpenEdXMobile/src/main/java/org/edx/mobile/task/EnqueueDownloadTask.java
package org.edx.mobile.task;

import android.content.Context;
import androidx.annotation.NonNull;

import com.google.inject.Inject;

import org.edx.mobile.model.db.DownloadEntry;
import org.edx.mobile.player.TranscriptManager;

import java.util.List;

public abstract class EnqueueDownloadTask extends Task<Long> {
    @Inject
    @NonNull
    TranscriptManager transcriptManager;
    @NonNull
    List<DownloadEntry> downloadList;

    public EnqueueDownloadTask(@NonNull Context context, @NonNull List<DownloadEntry> downloadList) {
        super(context);
        this.downloadList = downloadList;
    }

    @Override
    public Long call() {
        int count = 0;
        for (DownloadEntry downloadEntry : downloadList) {
            if (environment.getStorage().addDownload(downloadEntry) != -1) {
                count++;
                for (String value : downloadEntry.transcript.values()) {
                    transcriptManager.downloadTranscriptsForVideo(value, null);
                }
            }
        }
        return (long) count;
    }
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/task/EnqueueDownloadTask.kt
package org.edx.mobile.task

import android.content.Context
import dagger.hilt.android.EntryPointAccessors
import org.edx.mobile.core.EdxDefaultModule.ProviderEntryPoint
import org.edx.mobile.model.db.DownloadEntry

abstract class EnqueueDownloadTask(
    context: Context,
    private var downloadList: List<DownloadEntry>
) : Task<Long?>(context) {
    private var transcriptManager = EntryPointAccessors
        .fromApplication(context, ProviderEntryPoint::class.java).getTranscriptManager()

    override fun doInBackground(vararg params: Void?): Long {
        var count = 0
        for (downloadEntry in downloadList) {
            if (environment.storage.addDownload(downloadEntry) != -1L) {
                count++
                downloadEntry.transcript?.run {
                    for (value in this.values) {
                        transcriptManager.downloadTranscriptsForVideo(value, null)
                    }
                }
            }
        }
        return count.toLong()
    }
}
